Use EasyBuild information to get version and specifying paths to relevant dirs
Author: Viktor Rehnberg (Chalmers University of Technology)
diff -ruN a/setup.py b/setup.py
--- a/setup.py	2023-03-03 13:44:54.493375095 +0100
+++ b/setup.py	2023-03-03 15:33:08.481921491 +0100
@@ -2,10 +2,11 @@
 from distutils.extension import Extension
 from distutils.command.build_py import build_py as _build_py
 
-from os import system
+from os import path, environ
 import platform
 
 from distutils.core import setup, Extension
+from distutils.version import LooseVersion
 
 
 VERSION = "1.3.9"
@@ -22,7 +23,7 @@
 
 
 print("Looking for libOpenEXR...")
-if platform.system() == "Linux" and system("ldconfig -p | grep libOpenEXR"):
+if LooseVersion(environ['EBVERSIONOPENEXR']) < LooseVersion('3'):
     # There is no libOpenEXR, probably an old version of OpenEXR
     libraries=['Iex', 'Half', 'Imath', 'IlmImf', 'z']
 else:
@@ -44,15 +45,9 @@
   ext_modules=[ 
     Extension('OpenEXR',
               ['OpenEXR.cpp'],
-              include_dirs=['/usr/include/OpenEXR',
-                            '/usr/local/include/OpenEXR',
-                            '/opt/local/include/OpenEXR',
-                            '/usr/include/Imath',
-                            '/usr/local/include/Imath',
-                            '/opt/local/include/Imath'],
-              library_dirs=['/usr/lib',
-                            '/usr/local/lib',
-                            '/opt/local/lib'],
+              include_dirs=[path.join(environ['EBROOTOPENEXR'], 'include', 'OpenEXR'),
+                            path.join(environ['EBROOTOPENEXR'], 'include', 'Imath')],
+              library_dirs=[path.join(environ['EBROOTOPENEXR'], 'lib')],
               libraries=libraries,
               extra_compile_args=extra_compile_args)
   ],
